.github/workflows/build.yml

```
@@ -6,6 +6,8 @@ on:
     branches:    
       # Push events on main branch
       - main
```

For continuous feedback on PRs, I would remove this branch filter so that we can immediately see the effects of changes made. For most projects, I would make this a permanent change on `main` so that all branches run the build, as this will help catch build issues early and avoid them polluting main. (These cross-platform builds often get broken by accidentally non-cross-platform code, as well as any changes to the build config itself as in this PR.) See [this newsletter article](https://hashicorp.atlassian.net/wiki/spaces/ENSV/pages/2299233161/Feb+18+2022+-+Faster+Build+Issue+Feedback) on why we made this the general recommendation back in Feb: 

```suggestion
    branches:    
      # Push events on main branch
      - main
```

---

I think this is worth committing sooner rather than later so we can see the latest results of the build here in the PR @cipherboy, what do you think?

---

Ah that's fine, we had added `workflow_dispatch` to this PR for that, but if this is better, we should remove the `workflow_dispatch`.

I think the right change is to remove both, right? As I read it the suggestion is a no-op which isn't what we want, we want to remove it completely. I'll make these changes now that we've settled on them.